src/contract/table-cursor.ts

```diff
@@ -1,60 +1,52 @@
-import {API, Serializer} from '@greymass/eosio'
+import {ABI, ABIDef, API, APIClient, Serializer} from '@greymass/eosio'
 import {wrapIndexValue} from '../utils'
-import {Query, Table} from './table'
 
-interface TableCursorParams {
-    table: Table
-    tableParams: API.v1.GetTableRowsParams
+type TableRowParamsTypes =
+    | API.v1.GetTableRowsParams
+    | API.v1.GetTableRowsParamsKeyed
+    | API.v1.GetTableRowsParamsTyped
+
+interface TableCursorArgs {
+    abi: ABIDef
+    client: APIClient
+    params: TableRowParamsTypes
     maxRows?: number
-    next_key?: API.v1.TableIndexType | string
-    indexPositionField?: string
 }
 
-/**
- * Represents a cursor for a table in the blockchain. Provides methods for
- * iterating over the rows of the table.
- *
- * @typeparam TableRow The type of rows in the table.
- */
-export class TableCursor<TableRow> {
-    private table: Table
+const defaultParams = {
+    json: false,
+    limit: 1000,
+}
+
+export class TableCursor<RowType = any> {
+    readonly abi: ABI
+    readonly type: string
+    readonly params: TableRowParamsTypes
+    readonly client: APIClient
+
     private next_key: API.v1.TableIndexType | string | undefined
-    private tableParams: API.v1.GetTableRowsParams
+
     private endReached = false
-    private indexPositionField?: string
     private rowsCount = 0
     private maxRows: number = Number.MAX_SAFE_INTEGER
 
-    /**
```

Comment:
I know that these were far from perfect, but we will still want to have some JSDoc statements here, right?

Reply:
Yeah absolutely. I just kept seeing so many things changing - that I wanted to redo them all if this was the direction we wanted to go 👍